lib/ret/hub.ex

```diff
@@ -16,7 +16,7 @@ defmodule Ret.Hub do
   use Bitwise
 
   @schema_prefix "ret0"
-  @primary_key {:hub_id, :integer, []}
+  @primary_key {:hub_id, :id, autogenerate: true}
```

Comment:
Yeah, but I thought that's what the `:returning` option on `insert` was for. https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert/2-options

Seems odd to have Ecto seemingly autogenerate here and then define a default `next_id()` fragment in the migration.

Reply:
oh, i'm pretty sure this isn't instructing Ecto about autogeneration, its *telling* Ecto that this field is autogenerated by the database (which is true because of `next_id()`, but would also be true if we say used `serial` or another method to generate the id via a sequence) so by having this Ecto will know to implicitly include it as a `RETURNING` column in all relevant queries

Reply:
in my testing, making this change does what I'd expect, `next_id()` generates the new value and then that value actually shows up in the returned data structure.